08-CMSIS/platform/STM32P103/inc/uart.h

```diff
@@ -0,0 +1,11 @@
+#ifndef __UART_H_         
+#define __UART_H_
+
+
+void usart_init(void);
+void print_str(const char *str);
+
```

Comment:
Do you mean [OOP in C](http://stackoverflow.com/questions/351733/can-you-write-object-oriented-code-in-c)?

I think the we don't need this kind of encapsulation.
The reasons are as follows:

- I move "uart.h" into "core/inc" because we only need same interface for the os ,and leave the implementation for each platform.This is done in the later commit.
It is a very simple solution.


- If we encapsulate the "uart family" as you mentioned,we need to create "serial_io" instace in their "uart.c".This is because the dependent functions are "static". Therefore,when we need to access "serial_io" instace in the os,we need to include the "uart.c",and **all the static functions are exposed to the entire os**.
`Another perspective:`
If I create the "serial_io" instace in the "os.c",what all I need is the implementation from the device.Therefore,I also need **"usart_init" and "print_str" being exposed.This is as same as the behavior in first point above**. I think we don't need to this.